### PR TITLE
test(cxx indexer): ensure xrefs are created for proto extension fields

### PR DESCRIPTION
**Pull request by justbuchanan**
https://github.com/protocolbuffers/protobuf/pull/5803 adds annotations between extension ids in C++ generated code and their corresponding descriptors in .proto files. This PR adds a verifier test for the behavior.

We can wait to merge this until after the protobuf PR is merged.

### CODE REVIEW

File: external.bzl [hunk 1]
```diff
@@ -690,12 +690,21 @@ def kythe_dependencies():
...
     maybe(
-        http_archive,
-        name = "com_google_protobuf",
-        sha256 = "712715f5ac35637131f0d829ca7e0edaccab6fdeb33ecd3692ff24214ae5032f",
-        strip_prefix = "protobuf-de9e1a04a68af0c8c5f49121ebd7dd1a2fed37af",
-        urls = ["https://github.com/protocolbuffers/protobuf/archive/de9e1a04a68af0c8c5f49121ebd7dd1a2fed37af.zip"],
+        git_repository,
+        name="com_google_protobuf",
+        remote = "https://github.com/justbuchanan/protobuf",
+        commit = "annotate_cc_extensions",
```
Comment (shahms) [inline external.bzl]: tag? I didn't know commit worked this way

File: external.bzl [hunk 2]
```diff
@@ -690,12 +690,21 @@ def kythe_dependencies():
...
+    # maybe(
+    #     http_archive,
+    #     name = "com_google_protobuf",
+    #     sha256 = "712715f5ac35637131f0d829ca7e0edaccab6fdeb33ecd3692ff24214ae5032f",
+    #     strip_prefix = "protobuf-de9e1a04a68af0c8c5f49121ebd7dd1a2fed37af",
+    #     urls = ["https://github.com/protocolbuffers/protobuf/archive/de9e1a04a68af0c8c5f49121ebd7dd1a2fed37af.zip"],
+    # )
+
+    # TODO(justbuchanan): Use the above protobuf dep instead once
+    # https://github.com/protocolbuffers/protobuf/pull/5803 is merged.
```
Comment (shahms) [inline external.bzl]: Given our history with protobuf reviews, I'd rather wait until it is merged than maintain a branch indefinitely.

File: external.bzl (hunk 1 above)
Comment (justbuchanan) [reply external.bzl]: Switched to `tag`, but yeah they both seem to work the same.

File: external.bzl (hunk 2 above)
Comment (justbuchanan) [reply external.bzl]: Works for me. We can just let this PR sit until the protobuf one is merged.

**Comment by justbuchanan** [comment]
The protobuf PR has been merged: https://github.com/protocolbuffers/protobuf/pull/5803.

File: external.bzl [hunk 3]
```diff
@@ -691,9 +691,9 @@ def kythe_dependencies():
     maybe(
         http_archive,
         name = "com_google_protobuf",
-        sha256 = "b50be32ea806bdb948c22595ba0742c75dc2f8799865def414cf27ea5706f2b7",
-        strip_prefix = "protobuf-3.7.0",
-        urls = ["https://github.com/protocolbuffers/protobuf/archive/v3.7.0.zip"],
```
Comment (shahms) [inline external.bzl]: nit, optional: Add a reminder to update to the next tagged release when available?

**Pull request merged**